lib/system/web/src/keyboard_engine.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::{HashMap, BTreeSet};
+use std::cell::RefCell;
+use std::rc::Rc;
+use wasm_bindgen::JsCast;
+use wasm_bindgen::prelude::*;
+use itertools::Itertools;
+
+enum Action {
+    KeyUp,
+    KeyDown
+}
+
+type Bindings = HashMap<String, Box<dyn FnMut()>>;
+
+#[wasm_bindgen]
+pub struct KeyboardEngine {
+    bindings:        Rc<RefCell<Bindings>>,
+    _element:         web_sys::Element,
+    _prevent_default: bool,
+    //keydown_cb:    Closure<dyn FnMut(web_sys::KeyboardEvent)>,
+    //keyup_cb:      Closure<dyn FnMut(web_sys::KeyboardEvent)>,
+    _state:          Rc<RefCell<BTreeSet<String>>>
```

Comment:
Don't use leading underscores in struct fields names until they are phantom values. Moreover, we tend to make every field public until it could break API or be not secure to use - is it the case with these fields?

Reply:
Got it, in fact it's values that are not used, so I've masked them from linter